tools/visual-diff-ci/run_ci_script.py

```
@@ -131,22 +134,28 @@ def kill(self):
 
 def start_frontend() -> BgProcess:
+    # react-router-serve takes its host/port from the HOST/PORT environment
+    # variables, not CLI flags. Appending `--host/--port` to `pnpm run start`
+    # would be parsed by pnpm (or ignored by the server), not forwarded, so set
+    # them via the environment instead.
     configs = {
         "VITE_SITE_URL": "http://127.0.0.1:8000",
         "VITE_API_URL": "http://127.0.0.1:8000",
         "VITE_AUTH_BASE_URL": "http://127.0.0.1:8000",
         "VITE_AUTH_RETURN_URL": "http://127.0.0.1:8000",
+        "HOST": "0.0.0.0",
```

Ruff (0.15.15)

[error] Possible binding to all interfaces

(S104)